rama-net/src/address/domain.rs

```diff
@@ -182,7 +183,7 @@ impl<'a> TryFrom<&'a [u8]> for Domain {
 
     fn try_from(name: &'a [u8]) -> Result<Self, Self::Error> {
         if is_valid_name(name) {
-            Ok(Self(Cow::Owned(
+            Ok(Self(SmolStr::new(
                 String::from_utf8(name.to_vec()).context("convert domain bytes to utf-8 string")?,
```

Comment:
```suggestion
                std::str::from_utf8(name).context("convert domain bytes to utf-8 string")?,
```

to avoid a heap allocation just to pass it as reference to `SmolStr` :)